test/puppetlabs/puppetdb/scf/storage_test.clj

```diff
@@ -1219,8 +1219,14 @@
     (let [catalog (:empty catalogs)]
       (add-certname! "node1")
       (add-certname! "node2")
-      (replace-catalog! (assoc catalog :certname "node1") (-> 2 days ago))
-      (replace-catalog! (assoc catalog :certname "node2") (now))
+      (replace-catalog! (assoc catalog
+                               :certname "node1"
+                               :producer_timestamp (-> 2 days ago))
+                        (now))
+      (replace-catalog! (assoc catalog
+                               :certname "node2"
+                               :producer_timestamp (now))
+                        (now))
```

Comment:
The most correct thing to do here is just choose a time and compute things relative to it. Using `now` and `ago` is a race, but it's a super coarse grained one and ends up be very readable, so I'm ok with it.